src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java

```diff
@@ -230,12 +261,16 @@ public int compare(InputSplit a, InputSplit b) {
 
 		for (int i = 0; i < origSplits.size();) {
 			try {
-				i = addIndexedSplits      (origSplits, i, newSplits, cfg);
-			} catch (IOException e) {
-				i = addProbabilisticSplits(origSplits, i, newSplits, cfg);
+                                if (cfg.getBoolean(ENABLE_BAI_SPLIT_CALCULATOR, false)) {
+                                        i = addBAISplits    (origSplits, i, newSplits, cfg);
+                                } else {
+				        i = addIndexedSplits(origSplits, i, newSplits, cfg);
+                                }
```

Comment:
I don't think this is quite right. Even if `hadoopbam.bam.enable-bai-splitter` is true, if there is a splitting-bai file, then it should be used, since it has been generated for finding splits.

So I think the precedence order should be 1) use splitting-bai (if it exists), 2) using bai (if it exists and `hadoopbam.bam.enable-bai-splitter` is true), 3) use heuristics. If 1 or 2 fail, then fall back to 3 also.